### DISCUSSION TITLE
Reconnecting to a source

### DISCUSSION THREAD

**Discussion by creategui** [bluenviron/mediamtx#924] [Questions]
Hi,
I have several source rtsp cameras I am connecting to over a WAN and occasionally there will be a network hiccup that causes one or more "source" to become unavailable from rtsp-simple-server.  Is there a config parameter that will force it to reconnect to the source if there is no "publisher"?

My work around for now has been to restart the rtsp-simple-server container in a cron.daily script but this is far from optimal as it restarts all sources and interrupts the feeds that were not having issues and I may go a few hours without a source.

my sources are simply configured as:

  locationA-cam-01:
    source: rtsp://10.10.10.123:554/profile4/media.smp

**Comment by aler9** [comment]
Hello, reconnection to sources in case of disconnection is automatic and happens 5 seconds after disconnection. Can you post a server log?

**Reply by creategui** [reply]
I'm assuming you need a debug enabled log?

**Comment by creategui** [comment]
Some more info... I have seen the reconnection message in the logs.  So it does appear to be reconnecting when there are network issues.  In looking through the logs (not debug) it appears like rtsp-simple-server thinks it is connected but there is no data flowing through.  What I mean is that I am able to connect with ffmpeg and the server sees the connection but FFmpeg receives nothing.  Restarting the rtsp-simple-server fixes this.  I will enable the debug logs and see if anything shows up next time this problem shows itself.

[a comment by an automated account is not shown]
